python/ray/serve/controller.py

```diff
@@ -119,9 +121,30 @@ async def run_control_loop(self) -> None:
                     self.backend_state.update()
                 except Exception as e:
                     logger.error(f"Exception updating backend state: {e}")
-
+            self._put_serve_snapshot()
```

Comment:
This will be called 10x a second, is this reasonable?

Reply:
Should be fine. Really we should reduce the update period. The reason it's tuned so low right now is because we have a bunch of different states that we need to checkpoint between so deploying something takes a few update periods. cc @jiaodong we can increase the update period after your refactoring here.